### PR TITLE
Update tooling and fix deprecations

### PR DESCRIPTION
**Pull request by theodorejb**
* Update to PHPUnit 10 and require PHP 8.1+
* Fix coding standards with PHP-CS-Fixer
* Fix "Implicit conversion from float to int" deprecation warnings
* Fix incompatible return type deprecation warnings
* Fix passing null to string parameter deprecation warning
* Fix implicitly nullable parameter deprecation notices
* Add CI workflow

I didn't enable PHPStan as part of the CI workflow yet, since it currently fails due to missing `Parser::doCopy()` and `Parser::doUnion()` methods.

### CODE REVIEW

**Comment by nikic** [comment]
Thanks! I landed your CI workflow in https://github.com/ircmaxell/PHP-Yacc/commit/859157ccc2e9e2f8b07a89d107805ad565e72ac3 so we can see it running on this PR (it looks like GitHub doesn't pick up newly added workflows in PRs...)

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: PHP Composer
+on: [push, pull_request]
+jobs:
+  build:
+    name: Run tests on ${{ matrix.php }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php: [ '8.2', '8.3', '8.4' ]
```
Comment (nikic) [inline .github/workflows/php.yml]: We should check down to the minimum version in composer.json here, and use PHPUnit 10 to support it.

**Review by nikic — COMMENTED** [review]
This basically looks good to me. Only thing I'm concerned about is the raise in minimum PHP version.

Supporting PHP 7.0 doesn't make sense anymore, but I wonder if the minimum should stay at 7.4 to match PHP-Parser, which is the main user of this library. However, this is not strictly necessary because PHP-Yacc is only needed for development, not use, so I can see the argument in favor of the higher requirement.

File: .github/workflows/php.yml (hunk 1 above)
Comment (theodorejb) [reply .github/workflows/php.yml]: I rebased now and changed the first commit to require PHPUnit 10 and test on PHP 8.1+.

**Comment by theodorejb** [comment]
As long as the generated parser remains compatible with PHP 7.4, I think it's okay to have a higher requirement for development.

**Pull request merged**